manim/utils/tex_file_writing.py

```diff
@@ -42,67 +45,95 @@ def generate_tex_file(expression, tex_template, source_type):
     return result
 
 
-def tex_to_dvi(tex_file, use_ctex=False):
-    result = tex_file.replace(".tex", ".dvi" if not use_ctex else ".xdv")
+def tex_compilation_command(compiler, tex_file, tex_dir):
+    if (
+        compiler["command"] == "latex"
+        or compiler["command"] == "pdflatex"
+        or compiler["command"] == "luatex"
+        or compiler["command"] == "lualatex"
+    ):
+        commands = [
+            compiler["command"],
+            "-interaction=batchmode",
+            '-output-format="' + compiler["output_format"][1:] + '"',
+            "-halt-on-error",
+            '-output-directory="{}"'.format(tex_dir),
+            '"{}"'.format(tex_file),
+            ">",
+            os.devnull,
+        ]
+    elif compiler["command"] == "xelatex":
+        if compiler["output_format"] == ".xdv":
+            outflag = "-no-pdf"
+        elif compiler["output_format"] == ".pdf":
+            outflag = ""
+        else:
+            raise Exception("xelatex output is either pdf or xdv")
```

Comment:
Suggestion: always try to be as specific with Exceptions as possible. Maybe ValueError?